### ISSUE TITLE
Add Activation functions to juliamind.nn.functional.activation

### ISSUE THREAD

**Issue by diksha-shrivastava13** [diksha-shrivastava13/JuliaMind#1]
>  Please create a new issue with each of the subtasks and comment "- [ ] #issue_number" to keep the task list updated. 

- [ ] celu
- [ ] elu
- [ ] elu_
- [ ] gelu
- [ ] hardshrink
- [ ] hardtanh
- [ ] hardtanh_
- [ ] hardsigmoid
- [ ] hardswish
- [ ] leaky_relu
- [ ] leaky_relu_
- [ ] prelu
- [ ] rrelu
- [ ] relu
- [ ] relu_
- [ ] log_sigmoid
- [ ] maxout
- [ ] relu6
- [ ] selu
- [ ] silu
- [ ] softmax
- [ ] softmax_
- [ ] softplus
- [ ] softshrink
- [ ] softsign
- [ ] swish
- [ ] mish
- [ ] tanhshrink
- [ ] thresholded_relu
- [ ] thresholded_relu_
- [ ] log_softmax
- [ ] glu
- [ ] gumbel_softmax
